lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,1139 @@
+				'author',
+				'date',
+				'id',
+				'include',
+				'modified',
+				'parent',
+				'relevance',
+				'slug',
+				'include_slugs',
+				'title',
+			),
+		);
+
+		if ( 'page' === $this->post_type || post_type_supports( $this->post_type, 'page-attributes' ) ) {
+			$query_params['orderby']['enum'][] = 'menu_order';
+		}
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		if ( $post_type->hierarchical || 'attachment' === $this->post_type ) {
+			$query_params['parent']         = array(
+				'description' => __( 'Limit result set to items with particular parent IDs.' ),
+				'type'        => 'array',
+				'items'       => array(
+					'type' => 'integer',
+				),
+				'default'     => array(),
+			);
+			$query_params['parent_exclude'] = array(
+				'description' => __( 'Limit result set to all items except those of a particular parent ID.' ),
+				'type'        => 'array',
+				'items'       => array(
+					'type' => 'integer',
+				),
+				'default'     => array(),
+			);
+		}
+
+		$query_params['search_columns'] = array(
+			'default'     => array(),
+			'description' => __( 'Array of column names to be searched.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'enum' => array( 'post_title', 'post_content', 'post_excerpt' ),
+				'type' => 'string',
+			),
+		);
+
+		$query_params['slug'] = array(
+			'description' => __( 'Limit result set to posts with one or more specific slugs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'string',
+			),
+		);
+
+		$query_params['status'] = array(
+			'default'           => 'publish',
+			'description'       => __( 'Limit result set to posts assigned one or more statuses.' ),
+			'type'              => 'array',
+			'items'             => array(
+				'enum' => array_merge( array_keys( get_post_stati() ), array( 'any' ) ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => array( $this, 'sanitize_post_statuses' ),
+		);
+
+		$query_params = $this->prepare_taxonomy_limit_schema( $query_params );
+
+		if ( 'post' === $this->post_type ) {
+			$query_params['sticky'] = array(
+				'description' => __( 'Limit result set to items that are sticky.' ),
+				'type'        => 'boolean',
+			);
+		}
+
+		/*
+		* I have temporarilly removed this because I believe it is incorrect,
+		* if I don't comment this out, then there are no results if
+		* the query loop block is not set to filter by format.
+		$query_params['format'] = array(
+			'default'     => 'standard',
+			'description' => __( 'Limit result set to posts assigned one or more formats.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'anyOf' => array(
+					array(
+						'type' => 'string',
+						'enum' => array_values( get_post_format_slugs() ),
+					),
+					array(
+						'type'  => 'array',
+						'items' => array(
+							'type' => 'string',
+							'enum' => array_values( get_post_format_slugs() ),
+						),
+					),
+				),
+			),
+		);
+		*/
+
```

Comment:
This code does not seem to be needed, rather, it caused an issue where if I did not select any filters in the query loop block, there were no results. The expected result was for the latests posts to be displayed.

When removed/ commented out, the _filter_ still seems to work in the editor.

I don't know the REST API and the posts control well enough to know what updates are needed exactly, so this was part of my experimentation.